pkg/pillar/base/logobjecttypes.go

```diff
@@ -0,0 +1,443 @@
+	log "github.com/sirupsen/logrus"
+)
+
+// ObjectType : Predefined object types
+type ObjectType string
+
+// The following is NOT an extensive list.
+// Intent is to create a reference list for engineers to get an idea of its usage.
+const (
+	UnknownType  ObjectType = ""
+	LogType      ObjectType = "log"
+	RelationType ObjectType = "relation"
+)
+
+// LogObjectType :
+type LogObjectType string
+
+const (
+	// UnknownLogType : Invalid log type
+	UnknownLogType LogObjectType = ""
+	// ImageLogType :
+	ImageLogType LogObjectType = "image"
+	// NetworkInstanceLogType :
+	NetworkInstanceLogType LogObjectType = "network_instance"
+	// AppInstanceStatusLogType :
+	AppInstanceStatusLogType LogObjectType = "app_instance_status"
+	// AppInstanceConfigLogType :
+	AppInstanceConfigLogType LogObjectType = "app_instance_config"
+	// VolumeLogType :
+	VolumeLogType LogObjectType = "volume_type"
+)
+
+// RelationObjectType :
+type RelationObjectType string
+
+const (
+	// UnknownRelationType : Invalid relation type
+	UnknownRelationType RelationObjectType = ""
+	// AddRelationType :
+	AddRelationType RelationObjectType = "add_relation"
+	// DeleteRelationType :
+	DeleteRelationType RelationObjectType = "delete_relation"
+)
+
+// LogObject : Holds all key value pairs to be logged later.
+type LogObject struct {
+	Initialized bool
+	Fields      map[string]interface{}
+}
+
+var logObjectMap = make(map[string]*LogObject)
+
+// LoggableObject :
+type LoggableObject interface {
+	LogKey() string
+	LogCreate()
+	LogModify(old interface{})
+	LogDelete()
+}
+
+// NewLogObject :
+// objType -> [MANDATORY] log, action, relation, app_instance, image etc
+// objName -> App instance name, name of file being downloaded etc
+// objUUID -> UUID of the object if present (App instance UUID) or UUID of parent object
+// key     -> [MANDATORY] Key used for storing internal data. This should be the same Key your LoggableObject.Key()
+// would return. LogObject craeted here and the corresponding LoggableObject are linked using this key.
+// objType and objName are mandatory parameters
+func NewLogObject(objType LogObjectType, objName string, objUUID uuid.UUID, key string) *LogObject {
+	if objType == UnknownLogType || len(key) == 0 {
+		log.Fatal("NewLogObject: objType and key parameters mandatory")
+	}
+	// Check if we already have an object with the given key
+	object, ok := logObjectMap[key]
+	if ok {
+		return object
+	}
+
+	object = new(LogObject)
+	InitLogObject(object, objType, objName, objUUID, key)
+
+	return object
+}
+
+// InitLogObject : Initialize an already allocated LogObject
+func InitLogObject(object *LogObject, objType LogObjectType, objName string, objUUID uuid.UUID, key string) {
+	if objType == UnknownLogType || len(key) == 0 {
+		log.Fatal("InitLogObject: objType and key parameters mandatory")
+	}
+	if object == nil {
+		log.Fatal("InitLogObject: LogObject cannot be nil")
+	}
+	fields := make(map[string]interface{})
+	fields["main_type"] = LogType
```

Comment:
We should name this field based on the name of the enum (wheter that becomes LogEventType or something else)

Reply:
will do